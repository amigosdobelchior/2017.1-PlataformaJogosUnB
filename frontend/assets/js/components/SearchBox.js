import _ from 'lodash'
import React, { Component } from 'react'
import { Search, Grid } from 'semantic-ui-react'

export default class SearchBox extends Component {
    loadGameFromServer () {
        fetch("/api/list/",
            {
                "headers": new Headers({
                    "Accept": "application/json",
                    "Content-Type": "application/json",
                }),
                "method": "GET"
            }).
          then((response) => response.json()).
          then((games) => {
              this.setState({games});
          });
    }

    constructor(props) {
        super(props);
        this.resetComponent = this.resetComponent.bind(this)
        this.handleResultSelect = this.handleResultSelect.bind(this)
        this.handleSearchChange = this.handleSearchChange.bind(this)
    }

    componentWillMount() {
        this.loadGameFromServer()
        this.resetComponent()
    }

    resetComponent() {
        this.setState({ isLoading: false, results: [], value: '' })
    }

    handleResultSelect(e, result) {
        this.setState({ value: result.name })
        window.location = `/games/${result.pk}`;
    }

    handleSearchChange(e, value) {
        this.setState({ isLoading: true, value })

        setTimeout(() => {
            if (this.state.value.length < 1) return this.resetComponent()

            const re = new RegExp(_.escapeRegExp(this.state.value), 'i')
            const isMatch = (result) => re.test(result.name)

            this.setState({
                isLoading: false,
                results: _.filter(this.state.games, isMatch),
            })
        }, 500)
    }

    render() {
        const { isLoading, value, results } = this.state

        return (
            <Grid>
                <Grid.Column width={8}>
                    <Search
                        size='small'
                        placeholder='Search...'
                        loading={isLoading}
                        onResultSelect={this.handleResultSelect}
                        onSearchChange={this.handleSearchChange}
                        results={results}
                        value={value}
                        {...this.props}
                    />
                </Grid.Column>
            </Grid>
        )
    }
}