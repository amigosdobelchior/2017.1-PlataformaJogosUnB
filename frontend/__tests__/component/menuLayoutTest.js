import React from 'react';
import SearchBox from '../../assets/js/components/SearchBox'
import {Container, Grid, Header, Menu, Segment, Sidebar, Icon} from "semantic-ui-react";
import renderer from 'react-test-renderer';
var expect = required('expect');

test('Test render MenuLayout', () => {
    const activeItem = null;
  const component = renderer.create(

      <div>
                <Grid>
                    <Grid.Row only="tablet mobile">
                        <Grid.Column>
                            <Sidebar.Pusher>
                                <Sidebar as={Menu} animation="overlay" width="wide" visible="visible" vertical inverted onClick="active">
                                    <Menu.Item as="link" to="/" active={activeItem === "/"}><Header inverted>Home</Header></Menu.Item>
                                    <Menu.Item as="link" to="/games/" active={activeItem === "/games/"}><Header inverted>Jogos</Header></Menu.Item>
                                    <Menu.Item as="link" to="/about/" active={activeItem === "/about/"}><Header inverted>Sobre</Header></Menu.Item>
                                </Sidebar>
                                <Segment inverted>
                                    <Menu inverted pointing secondary>
                                        <Icon name='bars' size='big' onClick="active" />
                                    </Menu>
                                </Segment>
                            </Sidebar.Pusher>
                        </Grid.Column>
                    </Grid.Row>

                    <Grid.Row only="computer large">
                        <Grid.Column>
                            <Segment inverted stackable container>
                                <Menu inverted pointing secondary>
                                    <Container>
                                        <Menu.Item as="link" to="/" active={activeItem === "/"}><Header inverted>Home</Header></Menu.Item>
                                        <Menu.Item as="link" to="/games/" active={activeItem === "/games/"}><Header inverted>Jogos</Header></Menu.Item>
                                        <Menu.Item as="link" to="/about/" active={activeItem === "/about/"}><Header inverted>Sobre</Header></Menu.Item>
                                        <Menu.Menu position='right'>
                                            <Menu.Item>
                                                <SearchBox />
                                            </Menu.Item>
                                        </Menu.Menu>
                                    </Container>
                                </Menu>
                            </Segment>
                        </Grid.Column>
                    </Grid.Row>
                </Grid>
            </div>

      );



  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
