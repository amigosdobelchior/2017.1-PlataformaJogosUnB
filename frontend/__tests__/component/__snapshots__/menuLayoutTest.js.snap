// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render MenuLayout 1`] = `
<div>
  <div
    className="ui grid"
  >
    <div
      className="tablet only mobile only row"
    >
      <div
        className="column"
      >
        <div
          className="pusher"
        >
          <div
            className="ui inverted vertical ui overlay left wide visible sidebar menu"
            onClick="active"
          >
            <link
              className="item"
              onClick={[Function]}
              to="/"
            >
              <div
                className="ui inverted header"
              >
                Home
              </div>
            </link>
            <link
              className="item"
              onClick={[Function]}
              to="/games/"
            >
              <div
                className="ui inverted header"
              >
                Jogos
              </div>
            </link>
            <link
              className="item"
              onClick={[Function]}
              to="/about/"
            >
              <div
                className="ui inverted header"
              >
                Sobre
              </div>
            </link>
          </div>
          <div
            className="ui inverted segment"
          >
            <div
              className="ui inverted pointing secondary menu"
            >
              <i
                aria-hidden="true"
                className="bars big icon"
                onClick="active"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="computer only large only row"
    >
      <div
        className="column"
      >
        <div
          className="ui inverted segment"
          container={true}
          stackable={true}
        >
          <div
            className="ui inverted pointing secondary menu"
          >
            <div
              className="ui container"
            >
              <link
                className="item"
                onClick={[Function]}
                to="/"
              >
                <div
                  className="ui inverted header"
                >
                  Home
                </div>
              </link>
              <link
                className="item"
                onClick={[Function]}
                to="/games/"
              >
                <div
                  className="ui inverted header"
                >
                  Jogos
                </div>
              </link>
              <link
                className="item"
                onClick={[Function]}
                to="/about/"
              >
                <div
                  className="ui inverted header"
                >
                  Sobre
                </div>
              </link>
              <div
                className="right menu"
              >
                <div
                  className="item"
                  onClick={[Function]}
                >
                  <div
                    className="ui grid"
                  >
                    <div
                      className="eight wide column"
                    >
                      <div
                        className="ui small search"
                        onBlur={[Function]}
                        onFocus={[Function]}
                        onMouseDown={[Function]}
                      >
                        <div
                          className="ui icon input"
                        >
                          <input
                            autoComplete="off"
                            className="prompt"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            placeholder="Pesquisar..."
                            tabIndex="0"
                            type="text"
                            value=""
                          />
                          <i
                            aria-hidden="true"
                            className="search icon"
                          />
                        </div>
                        <div
                          className="results transition"
                        >
                          <div
                            className="message empty"
                          >
                            <div
                              className="header"
                            >
                              No results found.
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
